azure-quantum/tests/unit/test_workspace.py

```diff
@@ -13,95 +13,92 @@
-        resource_group=resource_group,
-        name=name)
-    assert ws.location == "westus"
-    
-    ws = Workspace(
-        subscription_id=subscription_id,
-        resource_group=resource_group,
-        name=name,
-        location="   ")
-    assert ws.location == "westus"
-    
-    # User-provided location name should be normalized
-    location = "East US"
-    ws = Workspace(
-        subscription_id=subscription_id,
-        resource_group=resource_group,
-        name=name,
-        location=location)
-    assert ws.location == "eastus"
-
-def test_create_workspace_instance_invalid():
-    subscription_id = "44ef49ad-64e4-44e5-a3ba-1ee87e19d3f4"
-    resource_group = "rg"
-    name = "n"
-    storage = "strg"
-
-    with pytest.raises(ValueError):
-        ws = Workspace()
-
-    with pytest.raises(ValueError):
         ws = Workspace(
             subscription_id=subscription_id,
             resource_group=resource_group,
-            name="")
+            name=name)
+        assert ws.subscription_id == subscription_id
+        assert ws.resource_group == resource_group
+        assert ws.name == name
+
+        ws = Workspace(
+            subscription_id=subscription_id,
+            resource_group=resource_group,
+            name=name,
+            storage=storage)
+        assert ws.storage == storage
+
+        resource_id = f"/subscriptions/{subscription_id}/RESOurceGroups/{resource_group}/providers/Microsoft.Quantum/Workspaces/{name}"
+        ws = Workspace(resource_id=resource_id)
+        assert ws.subscription_id == subscription_id
+        assert ws.resource_group == resource_group
+        assert ws.name == name
+
+        ws = Workspace(resource_id=resource_id, storage=storage)
+        assert ws.storage == storage
 
-    with pytest.raises(ValueError):
-        ws = Workspace(resource_id="invalid/resource/id")
+    def test_create_workspace_locations(self):
+        subscription_id = "44ef49ad-64e4-44e5-a3ba-1ee87e19d3f4"
+        resource_group = "rg"
+        name = "n"
 
-    with pytest.raises(ValueError):
-        ws = Workspace(storage=storage)
+        # Default should be westus
+        ws = Workspace(
+            subscription_id=subscription_id,
+            resource_group=resource_group,
+            name=name)
+        assert ws.location == "westus"
+        
+        ws = Workspace(
+            subscription_id=subscription_id,
+            resource_group=resource_group,
+            name=name,
+            location="   ")
+        assert ws.location == "westus"
+        
+        # User-provided location name should be normalized
+        location = "East US"
+        ws = Workspace(
+            subscription_id=subscription_id,
+            resource_group=resource_group,
+            name=name,
+            location=location)
+        assert ws.location == "eastus"
+
+    def test_create_workspace_instance_invalid(self):
+        subscription_id = "44ef49ad-64e4-44e5-a3ba-1ee87e19d3f4"
```

Comment:
Is this a secret or a dummy subscription ID? This will most likely raise a flag for the compliance tool so please add an inline comment to explain that it's a dummy ID that can be safely ignored.

Reply:
It was a dummy subscription id, but with PR #68 these literals were removed and now we use the base class properties.